ipp-v3-java-devkit/src/test/java/com/intuit/ipp/interceptors/CallbackHandlerInterceptorCDCTest.java

```diff
@@ -1,33 +1,227 @@
 package com.intuit.ipp.interceptors;
 
 import com.intuit.ipp.data.CDCResponse;
+import com.intuit.ipp.data.Error;
+import com.intuit.ipp.data.Fault;
+import com.intuit.ipp.data.IntuitEntity;
+import com.intuit.ipp.data.QueryResponse;
 import com.intuit.ipp.exception.FMSException;
 import com.intuit.ipp.services.CDCQueryResult;
 import org.testng.annotations.Test;
 
+import javax.xml.bind.JAXBElement;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
-
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertNotNull;
 import static org.testng.Assert.assertNull;
 
 public class CallbackHandlerInterceptorCDCTest extends CallbackHandlerBase {
 
-
     @Test
     public void  emptyListIsOk() throws FMSException {
-        final List<CDCQueryResult> results = invokeCDC(Collections.<CDCResponse>emptyList());
-        assertNull(results);
+        assertNull(invokeCDC(Collections.<CDCResponse>emptyList()));
+    }
+
+    @Test
+    public void  singleWithResponse() throws FMSException {
+        assertEmptyResult(invokeCDC(Collections.singletonList(new CDCResponse())));
+    }
+
+    @Test
+    public void  singleResponseWithEmptyQuery() throws FMSException {
+        final CDCResponse o = new CDCResponse();
+        o.setQueryResponse(Collections.singletonList(new QueryResponse()));
+
+        assertEmptyResult(invokeCDC(Collections.singletonList(o)));
+    }
+
+    @Test
+    public void  singleEntityIsOk() throws FMSException {
+        final CDCResponse response = new CDCResponse();
+        final QueryResponse queryResponse = new QueryResponse();
+
+        queryResponse.setIntuitObject(Collections.<JAXBElement<? extends IntuitEntity>>singletonList(getDummyTestEntity()));
+
+        response.setQueryResponse(Collections.singletonList(queryResponse));
+
+        new ResultChecker( assertAndGetFirst(invokeCDC(Collections.singletonList(response))))
+            .assertQueryKeys("IntuitTestEntity");
     }
 
+
+    /**
+     * Illustrates https://github.com/intuit/QuickBooks-V3-Java-SDK/issues/75
+     * Error is expected in result, but it gives empty result.
+     *
+     * @throws FMSException
+     */
     @Test
-    public void  singleItemIsOk() throws FMSException {
-        final List<CDCQueryResult> results = invokeCDC(Collections.singletonList(new CDCResponse()));
+    public void  errorNoResponse() throws FMSException {
+        final CDCResponse o = new CDCResponse();
+        final Fault fault = new Fault();
+        final Error error = new Error();
+        fault.setError(Collections.singletonList(error));
+        error.setDetail("My custom error");
+        o.setFault(fault);
+        o.setQueryResponse(null);
 
-        assertNotNull(results);
-        assertFalse(results.isEmpty());
-        assertEquals(1, results.size());
+        assertEmptyResult(invokeCDC(Collections.singletonList(o)));
```

Comment:
I think this uncovers either dead code in callback interceptor or potential defect. My expectation here is to be able to get error back.